fabric-content-registries-v0/src/main/java/net/fabricmc/fabric/api/registry/OxidizableBlocksRegistry.java

```diff
@@ -0,0 +1,213 @@
+	private static final BiMap<Block, Block> UNWAXED_TO_WAXED_BLOCKS = HashBiMap.create();
+	private static final BiMap<Block, Block> WAXED_TO_UNWAXED_BLOCKS = HashBiMap.create();
+
+	static {
+		// Put all vanilla entries in here, just in case!
+		OXIDIZATION_LEVEL_INCREASES.putAll(Oxidizable.OXIDATION_LEVEL_INCREASES.get());
+		OXIDIZATION_LEVEL_DECREASES.putAll(Oxidizable.OXIDATION_LEVEL_DECREASES.get());
+		UNWAXED_TO_WAXED_BLOCKS.putAll(HoneycombItem.UNWAXED_TO_WAXED_BLOCKS.get());
+		WAXED_TO_UNWAXED_BLOCKS.putAll(HoneycombItem.WAXED_TO_UNWAXED_BLOCKS.get());
+	}
+
+	/**
+	 * Registers multiple {@link OxidizableFamily}s.
+	 * @param families the families to register
+	 */
+	public static void registerFamilies(OxidizableFamily... families) {
+		for (OxidizableFamily family : families) {
+			registerFamily(family);
+		}
+	}
+
+	/**
+	 * Registers an {@link OxidizableFamily}s.
+	 * @param family the {@link OxidizableFamily} to register
+	 */
+	public static void registerFamily(OxidizableFamily family) {
+		OXIDIZATION_LEVEL_INCREASES.putAll(family.oxidizationLevelIncreasesMap());
+		OXIDIZATION_LEVEL_DECREASES.putAll(family.oxidizationLevelDecreasesMap());
+		UNWAXED_TO_WAXED_BLOCKS.putAll(family.unwaxedToWaxedMap());
+		WAXED_TO_UNWAXED_BLOCKS.putAll(family.waxedToUnwaxedMap());
+	}
+
+	/**
+	 * Registers a waxable block.
+	 * Unnecessary if part of a registered {@link OxidizableFamily}.
+	 * @see #registerFamily(OxidizableFamily)
+	 * @param unwaxed the unwaxed variant
+	 * @param waxed the waxed variant
+	 */
+	public static void registerWaxable(Block unwaxed, Block waxed) {
+		UNWAXED_TO_WAXED_BLOCKS.put(unwaxed, waxed);
+		WAXED_TO_UNWAXED_BLOCKS.put(waxed, unwaxed);
+	}
+
+	/**
+	 * Gets the map of {@link Oxidizable.OxidizationLevel} increases.
+	 * @return the map
+	 */
+	public static BiMap<Block, Block> getOxidizationLevelIncreases() {
+		return OXIDIZATION_LEVEL_INCREASES;
+	}
+
+	/**
+	 * Gets the map of {@link Oxidizable.OxidizationLevel} decreases.
+	 * @return the map
+	 */
+	public static BiMap<Block, Block> getOxidizationLevelDecreases() {
+		return OXIDIZATION_LEVEL_DECREASES;
+	}
+
+	/**
+	 * Gets the map of unwaxed blocks to waxed counterparts.
+	 * @return the map
+	 */
+	public static BiMap<Block, Block> getUnwaxedToWaxedBlocks() {
+		return UNWAXED_TO_WAXED_BLOCKS;
+	}
+
+	/**
+	 * Gets the map of waxed blocks to unwaxed counterparts.
+	 * @return the map
+	 */
+	public static BiMap<Block, Block> getWaxedToUnwaxedBlocks() {
+		return WAXED_TO_UNWAXED_BLOCKS;
+	}
+
+	/**
+	 * Gets an Optional of the {@link Block} that would result from a decrease in {@link Oxidizable.OxidizationLevel} from the given block.
+	 * If no such block exists, the Optional will be empty.
+	 * @param block the block whose oxidization is to be decreased
+	 * @return the block with decreased oxidization, in an optional
+	 */
+	public static Optional<Block> getDecreasedOxidizationBlock(Block block) {
```

Comment:
Please use `@Nullable`, no `Optional`.

Reply:
Optional is how the vanilla stuff does it, I'm sticking as close to the vanilla implementation as possible.